.pre-commit-config.yaml

```diff
@@ -17,27 +16,43 @@ repos:
           ]
         exclude: ".*(.fits|.fts|.fit|.txt|tca.*|extern.*|.rst|.md|docs/conf.py)$"
   - repo: https://github.com/charliermarsh/ruff-pre-commit
-    rev: "v0.4.5"
+    rev: "v0.3.5"
     hooks:
       - id: ruff
-        args: ["--fix", "--unsafe-fixes"]
-      - id: ruff-format
+        args: ["--fix"]
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.13.2
+    hooks:
+      - id: isort
+        exclude: ".*(.fits|.fts|.fit|.header|.txt|tca.*|extern.*|sunkit_image/extern)$"
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v4.6.0
     hooks:
       - id: check-ast
       - id: check-case-conflict
       - id: trailing-whitespace
+        exclude: ".*(.fits|.fts|.fit|.txt|.header|.csv)$"
       - id: mixed-line-ending
+        exclude: ".*(.fits|.fts|.fit|.txt|.csv)$"
       - id: end-of-file-fixer
+        exclude: ".*(.fits|.fts|.fit|.txt|.csv)$"
       - id: check-yaml
       - id: debug-statements
-  - repo: https://github.com/crate-ci/typos
-    rev: v1.21.0
+      - id: check-added-large-files
+        args: ["--enforce-all", "--maxkb=1054"]
+      - id: end-of-file-fixer
+        exclude: ".*(.fits|.fts|.fit|.header|.txt|tca.*|.json)$|^CITATION.rst$"
+      - id: mixed-line-ending
+        exclude: ".*(.fits|.fts|.fit|.header|.txt|tca.*)$"
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.2.6
     hooks:
-      - id: typos
```

Comment:
@nabobalis how's typos been for you? Do you think the experiment is a success and we should migrate the template to it?

Reply:
1) Faster
2) Better at finding words
3) Autofixes by default which can catch people out when adding it
4) Hates some normal words and requires some specific regex to avoid it replacing lines of code.

Reply:
so that sounds like a yes it is worth moving over, or are the regexing awful?

Reply:
Regex is fine, it's just tends to be specific to a word if you see https://github.com/sunpy/sunpy/blob/main/_typos.toml 

There probably is a better way to use their config but it was simplest to implement.